bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/AVMFritzBaseThingHandler.java

```diff
@@ -78,15 +94,202 @@ public void initialize() {
+    }
+
+    /**
+     * Updates thing channels.
+     *
+     * @param channelId ID of the channel to be updated.
+     * @param state State to be set.
+     */
+    private void updateThingChannelState(String channelId, State state) {
+        Channel channel = thing.getChannel(channelId);
+        if (channel != null) {
+            updateState(channel.getUID(), state);
+        } else {
+            logger.debug("Channel '{}' in thing '{}' does not exist, recreating thing.", channelId, thing.getUID());
+            createChannel(channelId);
+        }
+    }
+
+    /**
+     * Creates new channels for the thing.
+     *
+     * @param channelId ID of the channel to be created.
+     */
+    private void createChannel(String channelId) {
+        ThingHandlerCallback callback = getCallback();
+        if (callback != null) {
+            ChannelUID channelUID = new ChannelUID(thing.getUID(), channelId);
+            ChannelTypeUID channelTypeUID = CHANNEL_BATTERY.equals(channelId)
+                    ? new ChannelTypeUID("system:battery-level")
+                    : new ChannelTypeUID(BINDING_ID, channelId);
+            Channel channel = callback.createChannelBuilder(channelUID, channelTypeUID).build();
+            updateThing(editThing().withoutChannel(channelUID).withChannel(channel).build());
+        }
+    }
+
     /**
-     * Called from {@link AVMFritzBaseBridgeHandler)} to update the thing status because updateStatus is protected.
+     * Triggers thing channels.
      *
-     * @param status Thing status
-     * @param statusDetail Thing status detail
-     * @param description Thing status description
+     * @param channelId ID of the channel to be triggered.
+     * @param event Event to emit
      */
-    public void setStatusInfo(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
-        updateStatus(status, statusDetail, description);
+    private void triggerThingChannel(String channelId, String event) {
+        Channel channel = thing.getChannel(channelId);
+        if (channel != null) {
+            triggerChannel(channel.getUID(), event);
+        } else {
+            logger.debug("Channel '{}' in thing '{}' does not exist.", channelId, thing.getUID());
+        }
+    }
+
+    /**
+     * Updates thing channel configurations.
+     *
+     * @param channelId ID of the channel which configuration to be updated.
+     * @param configId ID of the configuration to be updated.
+     * @param value Value to be set.
+     */
+    private void updateThingChannelConfiguration(String channelId, String configId, Object value) {
+        Channel channel = thing.getChannel(channelId);
+        if (channel != null) {
+            Configuration config = channel.getConfiguration();
+            Configuration editConfig = new Configuration(new HashMap<>(config.getProperties()));
+            editConfig.put(configId, value);
+            config.setProperties(editConfig.getProperties());
```

Comment:
Is there a reason this isn't simplified?
```suggestion
            config.put(configId, value);
```